QtCollider/widgets/QcWebView.cpp

```diff
@@ -20,126 +20,226 @@
+
+  connect (page, SIGNAL(geometryChangeRequested(const QRect&)),
+           this, SIGNAL(geometryChangeRequested(const QRect&)));
+  
+  connect (page, SIGNAL(windowCloseRequested()),
+           this, SIGNAL(windowCloseRequested()));
+  
+  connect (page, SIGNAL(scrollPositionChanged(const QPointF&)),
+           this, SIGNAL(scrollPositionChanged(const QPointF&)));
+
+  connect (page, SIGNAL(contentsSizeChanged(const QSizeF&)),
+           this, SIGNAL(contentsSizeChanged(const QSizeF&)));
+
+  connect (page, SIGNAL(audioMutedChanged(bool)),
+           this, SIGNAL(audioMutedChanged(bool)));
+
+  connect (page, SIGNAL(recentlyAudibleChanged(bool)),
+           this, SIGNAL(recentlyAudibleChanged(bool)));
+
+  connect (page, SIGNAL(navigationRequested(QUrl,QWebEnginePage::NavigationType,bool)),
+           this, SLOT(onLinkClicked(QUrl,QWebEnginePage::NavigationType,bool)));
+
+  connect (page->action(QWebEnginePage::Reload), SIGNAL(triggered(bool)),
+          this, SLOT(onPageReload()) );
+
+  connect (page, SIGNAL(renderProcessTerminated(RenderProcessTerminationStatus, int)),
+          this, SLOT(onRenderProcessTerminated(RenderProcessTerminationStatus, int)) );
+}
+
+void WebView::onRenderProcessTerminated(QWebEnginePage::RenderProcessTerminationStatus status, int code)
+{
+  Q_EMIT(renderProcessTerminated((int)status, code));
+}
+  
+void WebView::triggerPageAction( int action, bool checked )
+{
+  QWebEngineView::triggerPageAction((QWebEnginePage::WebAction)action, checked);
 }
 
 QString WebView::url() const
 {
-  return QWebView::url().toString();
+  return QWebEngineView::url().toString();
 }
 
 void WebView::setUrl( const QString & str )
 {
   load( urlFromString(str) );
 }
 
-QString WebView::html () const
+bool WebView::delegateReload() const
+{
+  WebPage *p = qobject_cast<QtCollider::WebPage*>(page());
+  Q_ASSERT(p);
+  return p->delegateReload();
+}
+
+void WebView::setDelegateReload( bool flag )
 {
-  return page()->mainFrame()->toHtml();
+  WebPage *p = qobject_cast<QtCollider::WebPage*>(page());
+  Q_ASSERT(p);
+  p->setDelegateReload( flag );
 }
 
+void WebView::setFontFamily( int generic, const QString & specific )
+{
+  settings()->setFontFamily( (QWebEngineSettings::FontFamily) generic, specific );
+}
+  
+QAction* WebView::pageAction( QWebEnginePage::WebAction action) const
+{
+  return QWebEngineView::pageAction(action);
+}
+  
 void WebView::setHtml ( const QString &html, const QString &baseUrl )
 {
-  QUrl url( baseUrl.isEmpty() ? QUrl() : urlFromString(baseUrl) );
-  QWebView::setHtml( html, url );
+  if (page()) {
+    page()->setHtml(html, baseUrl);
+}
 }
 
-QString WebView::plainText () const
+void WebView::setContent(const QVector<int>& data, const QString& mimeType, const QString& baseUrl)
 {
-  return page()->mainFrame()->toPlainText();
+  if (page()) {
+    QByteArray byteData;
+    size_t i = 0;
+    for (int val : data) {
+      byteData.push_back((char)val);
+}
```

Comment:
`i` is unused. `}` on this line is misaligned. this is also memory-inefficient, you could call `byteData.reserve(data.size())` to get the right amount of memory immediately.